src/sipsess/update.c

```diff
@@ -149,20 +151,25 @@ int sipsess_update(struct sipsess *sess)
 	struct sipsess_request *req;
 	int err;
 
-	if (!sess || sess->terminated || !sess->ctype || !sess->desc)
+	if (!sess || sess->terminated || !sess->ctype || !sess->desc
+	   || (!sess->established && !sess->refresh_allowed)
+	   || sess->awaiting_answer)
 		return EINVAL;
```

Comment:
Maybe we can split up these checks in EINVAL and EPROTO error parts? Makes debugging sometimes easier.

Reply:
Makes sense, I split it up.